### PR TITLE
Use proper service names

### PR DESCRIPTION
**Pull request by mattBrzezinski**
### Resolves: #335, #408

This is an addendum to #336, as it didn't really fix the issue. Currently we are trying to use one piece of information to both create the URL which to hit and sign the request. This is not enough. We need to have both the `endpointPrefix` and the `signingName`.

This PR creates functionality to store both of these pieces of information and to use them appropriate. I should note that not all services have a `signingName`. In these cases we will use the `endpointPrefix` in lieu.

### CODE REVIEW

**Comment by mattBrzezinski** [comment]
bors try

**Comment by mattBrzezinski** [comment]
bors try

File: src/api_generation/utilities.jl [hunk 1]
```diff
@@ -239,3 +239,11 @@ function _get_function_parameters(input::String, shapes::AbstractDict{String})
...
     return (sort(required_parameters), sort(optional_parameters))
 end
+
+function _get_service_name(service::AbstractDict)
+    if haskey(service, "signingName") && service["signingName"] == service["endpointPrefix"]
+        return service["signingName"]
+    end
+
+    return service["endpointPrefix"]
+end
```
Comment (glennmoy) [inline src/api_generation/utilities.jl:249]: why do we need this function?
why not just return `service["endpointPrefix"]` in all cases?

Comment (mattBrzezinski) [reply src/api_generation/utilities.jl:249]: That makes sense... I actually just found an issue here. I'm revising it currently. Ignore this PR for the moment!

**Comment by mattBrzezinski** [comment]
Okay. Revised this, and we're set.

bors try

File: src/AWS.jl [hunk 2]
```diff
@@ -135,31 +135,35 @@ macro service(module_name::Symbol)
...
-    name::String
+    signing_name::String
+    endpoint_prefix::String
     api_version::String
 
     service_specific_headers::LittleDict{String, String}
 end
 
-RestJSONService(name::String, api_version::String) = RestJSONService(name, api_version, LittleDict{String, String}())
+RestJSONService(signing_name::String, endpoint_prefix::String, api_version::String) = RestJSONService(signing_name, endpoint_prefix, api_version, LittleDict{String, String}())
```
Comment (iamed2) [inline src/AWS.jl:166]: Could this be a kwarg? Could make src/AWSServices.jl cleaner. 

```julia
RestJSONService(signing_name::String, api_version::String; endpoint_prefix::String=signing_name) = RestJSONService(signing_name, endpoint_prefix, api_version, LittleDict{String, String}())
```

**Comment by glennmoy** [comment]
do you need to bump the patch version for this as well?

**Comment by mattBrzezinski** [comment]
> do you need to bump the patch version for this as well?

Yes. I tend to do those in separate PRs as in the past I've stepped on toes and ended up accidentally trying to skip versions which JuliaRegistrator gets upset about.

**Review by iamed2 — APPROVED** [review]

**Pull request merged**